### PR TITLE
COMMERCE 2580 1

### PR DESCRIPTION
**Pull request by ivicac**

### CODE REVIEW

**Comment by ivicac** [comment]
ci:test:sf

**Comment by ivicac** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-shipping-engine-remote/src/main/java/com/liferay/commerce/shipping/engine/remote/internal/RemoteCommerceShippingEngine.java [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+public class RemoteCommerceShippingEngine implements CommerceShippingEngine {
+
+	public static final String KEY = "remote";
+
+	@Override
+	public String getCommerceShippingOptionLabel(String name, Locale locale) {
+		throw new UnsupportedOperationException();
+	}
+
+	// TODO Ask Marco about context commerceOrder vs commerceOrder
```
Comment (dsitu) [inline modules/apps/commerce/commerce-shipping-engine-remote/src/main/java/com/liferay/commerce/shipping/engine/remote/internal/RemoteCommerceShippingEngine.java:92]: we should remove this comment.

```suggestion
```

File: modules/apps/commerce/commerce-shipping-engine-remote/src/main/java/com/liferay/commerce/shipping/engine/remote/internal/portlet/action/EditRemoteCommerceShippingMethodConfigurationMVCActionCommand.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Ivica Cardic
+ */
+@Component(
+	enabled = false, immediate = true,
+	property = {
+		"javax.portlet.name=" + CommercePortletKeys.COMMERCE_SHIPPING_METHODS,
+		"mvc.command.name=editRemoteCommerceShippingMethodConfiguration"
```
Comment (dsitu) [inline modules/apps/commerce/commerce-shipping-engine-remote/src/main/java/com/liferay/commerce/shipping/engine/remote/internal/portlet/action/EditRemoteCommerceShippingMethodConfigurationMVCActionCommand.java:43]: New mvc.command.name format https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/checks/mvc_command_name_check.markdown

/{javax.portlet.name}/{mvcActionCommand}
```suggestion
		"mvc.command.name=/commerce_shipping/edit_remote_commerce_shipping_method_configuration"
```

File: modules/apps/commerce/commerce-shipping-engine-remote/src/main/java/com/liferay/commerce/shipping/engine/remote/internal/RemoteCommerceShippingEngine.java [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+		_poolingHttpClientConnectionManager =
+			new PoolingHttpClientConnectionManager();
+
+		_poolingHttpClientConnectionManager.setMaxTotal(20);
+		_poolingHttpClientConnectionManager.setValidateAfterInactivity(30000);
+
+		HttpClientBuilder httpClientBuilder = HttpClients.custom();
+
+		httpClientBuilder.setConnectionManager(
+			_poolingHttpClientConnectionManager);
```
Comment (dsitu) [inline modules/apps/commerce/commerce-shipping-engine-remote/src/main/java/com/liferay/commerce/shipping/engine/remote/internal/RemoteCommerceShippingEngine.java:143]: Wrap like burrito.
```suggestion
		HttpClientBuilder httpClientBuilder = HttpClients.custom();

		_poolingHttpClientConnectionManager =
			new PoolingHttpClientConnectionManager();

		_poolingHttpClientConnectionManager.setMaxTotal(20);
		_poolingHttpClientConnectionManager.setValidateAfterInactivity(30000);

		httpClientBuilder.setConnectionManager(
			_poolingHttpClientConnectionManager);
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by ivicac** [comment]
ci:test:sf

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by marco-leo** [comment]
ci:forward

**Comment by marco-leo** [comment]
@riccardo-alberti  FYI

**Pull request closed without merging**